### PR TITLE
Project skill

### LINKED ISSUE #116: Add Project Skills m2m

### PR DESCRIPTION
**Pull request by snewcomer**
Closes #116.

### CODE REVIEW

**Comment by joshsmith** [comment]
@begedin would you mind reviewing when you have a chance? I'm not sure what I missed here.

File: test/controllers/project_skill_controller_test.exs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+    assert json_response(conn, 422)["errors"] != %{}
+  end
+
+  test "deletes chosen resource", %{conn: conn} do
+    project_skill = Repo.insert! %ProjectSkill{}
+    conn = delete conn, project_skill_path(conn, :delete, project_skill)
+    assert response(conn, 204)
+    refute Repo.get(ProjectSkill, project_skill.id)
+  end
```
Comment (begedin) [inline test/controllers/project_skill_controller_test.exs:125]: Missing a 404 when trying to delete a non existent project skill spec. I see we're not really consistent in writing this spec for other such resources.

`OrganizationMembershipController` and `ProjectCategoryController` have it:

https://github.com/code-corps/code-corps-phoenix/blob/develop/test/controllers/project_category_controller_test.exs#L81

`RoleSkillController`, `UserCategoryController`, `UserSkillController`, `UserRoleController` do not have it.

@JoshSmith Do we want to add an issue to make this consistent? Either we put the spec everywhere, or we remove it.

**Comment by joshsmith** [comment]
@begedin yes, let's write an issue to keep it consistent.

File: test/controllers/project_skill_controller_test.exs (hunk 1 above)
Comment (begedin) [reply test/controllers/project_skill_controller_test.exs:125]: Ok, so we need to add a spec for attempting to delete a non-existant record, in a style like the one at 

https://github.com/code-corps/code-corps-phoenix/blob/develop/test/controllers/project_category_controller_test.exs#L81

I've created an issue for the other ones - #151

**Comment by begedin** [comment]
Good to go after adding the missing spec 👍

**Comment by joshsmith** [comment]
@snewcomer will you have a chance today to add this in?

**Comment by joshsmith** [comment]
👍

**Pull request merged**